src/JsonSchema/Constraints/ConstraintInterface.php

```diff
@@ -33,10 +33,12 @@ public function addErrors(array $errors);
     /**
      * adds an error
      *
-     * @param $path
-     * @param $message
+     * @param string $path
+     * @param string $message
+     * @param string $constraint the constraint/rule that is broken, e.g.: 'minLength'
+     * @param array $more more array elements to add to the error
      */
-    public function addError($path, $message);
+    public function addError($path, $message, $constraint='', $more=null);
```

Comment:
It's in ConstraintInterface.php the phpdoc here is just inheriting the interface class.